src/include/daos_srv/ras.h

```diff
@@ -25,53 +25,72 @@
  * RAS event definitions to the used in either data or control planes.
  */
 
-#ifndef __RAS_H__
-#define __RAS_H__
+#ifndef __DAOS_RAS_H__
+#define __DAOS_RAS_H__
 
-#define RAS_ID_UNKNOWN_STR "Unknown RAS event"
-#define RAS_SEV_UNKNOWN_STR "Unknown RAS event severity"
-#define RAS_TYPE_UNKNOWN_STR "Unknown RAS event type"
+#define DAOS_RAS_STR_FIELD_SIZE 128
+#define DAOS_RAS_ID_FIELD_SIZE 64
 
-enum ras_event_id {
-	RAS_RANK_EXIT	= 1,
-	RAS_RANK_NO_RESP,
-};
+/**
+ * For each RAS event, define the following:
+ * - Enum symbol to use in the code to identify the RAS event
+ *   No external visibility.
+ * - 64-char string identifier raised as part of the event
+ *   The identifier just be prefixed by component_
+ *   Carried over with the RAS event.
+ */
+#define RAS_EVENT_LIST						\
+	X(RAS_RANK_UP,		"engine_status_up")		\
+	X(RAS_RANK_DOWN,	"engine_status_down")		\
+	X(RAS_RANK_NO_RESPONSE,	"engine_status_no_response")	\
+	X(RAS_POOL_REPS_UPDATE,	"pool_replicas_updated")
 
-static inline char *
-ras_event_id_enum_to_name(enum ras_event_id id)
-{
-	switch (id) {
-	case RAS_RANK_EXIT:
-		return "daos_rank_exited";
-	case RAS_RANK_NO_RESP:
-		return "daos_rank_no_response";
-	}
+/** Define RAS event enum */
+typedef enum {
+#define X(a, b) a,
+	RAS_EVENT_LIST
+#undef X
+} ras_event_t;
 
-	return RAS_ID_UNKNOWN_STR;
+/** Extract RAS event ID (= 64-char string) from enum */
+static inline char *
+ras_event2str(ras_event_t ras) {
+#define X(a, b) case a: return b;
```

Comment:
(style)  Macros with multiple statements should be enclosed in a do - while loop
(style)  Macros with flow control statements should be avoided
(style)  macros should not use a trailing semicolon